packages/push/src/PushRegistration.ts

```diff
@@ -19,13 +19,19 @@ export class PushRegistration {
   public static readonly TYPE: string = "push";
   public static readonly API_PATH: string = "rest/registry/device";
 
-  public pushConfig: any;
+  private pushConfig: ServiceConfiguration;
 
   constructor() {
     const configuration = INSTANCE.getConfigByType(PushRegistration.TYPE);
     if (!configuration || configuration.length === 0) {
       console.warn("Push configuration is missing. UPS server registration will not work.");
-      this.pushConfig = {};
+      this.pushConfig = {
```

Comment:
Ah.. sorry I should have said this before, it makes no sense setting a config if none is found. The property should be conditional and left it empty in this case. If it's initialized here, `hasConfig` will return always `true`.
Take a look at [MetricsService](https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/core/src/metrics/MetricsService.ts#L24)

Reply:
@josemigallas good catch, thanks :+1: